src/service/viewer-impl.js

```diff
@@ -362,6 +363,27 @@ export class Viewer {
       }
     });
 
+    // Replace URL if requested.
+    const replaceUrlParam = this.params_['replaceUrl'];
+    if (ampdoc.isSingleDoc() &&
+        replaceUrlParam &&
+        this.win.history.replaceState) {
```

Comment:
I believe this is still an issue. Last time we tried to remove this check, we saw lots of errors. Feel free to file a bug for a deeper cleanup. But it will have to first start in the `history-impl.js` with error sampling for when it's not available.